### PR TITLE
Add tests, CI, and README

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Automated tests will help to reduce the risk that we ever intentionally break the map (aka "Continous Integration" or CI). Those will run using GitHub Actions every time someone opens a PR or merges a PR. They can only merge their PR once the tests are working ("green").

People can also run these tests locally by using `npm test`. 

This also adds some other commands:

* `npm start`, which will use `http-server` to start the site. (While you could simply open up `index.html`, it won't properly load all the assets so the map will look empty.)
* `npm fmt` to run an autoformatter called Prettier, which formats our code like fixing indentation. 
* `npm lint` to run Prettier in check-only mode and to run a tool called ESLint that checks for common issues/bugs.

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+env:
```
Comment (Eric-Arellano) [inline .eslintrc.yml:1]: This is config for the ESLint tool. It was auto-generated and uses the defaults.

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      uses: actions/checkout@v2
+    - name: Set up Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: 18
+    - name: Install dependencies
+      run: npm ci
+    - name: Start server
+      run: |
+        npm start &
```
Comment (Eric-Arellano) [inline .github/workflows/tests.yaml:18]: This means to start the server in the background rather than interactively.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,52 @@
...
+The code is fairly simple and we intentionally are not using fancy frameworks like React or Svelte. The main files are `index.html`, `js/script.js`, and `data/*.geojson`. `js/script.js` will load the `.geojson` files to dynamically update `index.html` with all our data.
+
+# How tos
+
+All the commands take place in a _terminal_, a text-based interface for interacting with your computer. On macOS, you can open the "Terminal" app. On Windows, you can use "Git Bash", which will be installed when you install Git below. 
+
+You will also need to install:
+
+* Git, which we use for "version control": https://git-scm.com/book/en/v2/Getting-Started-Installing-Git
+* NPM, which is how we run JavaScript and our tests. Use the LTS at https://nodejs.dev/en/download/
```
Comment (Eric-Arellano) [inline README.md:14]: I think it's going to be too hard to explain every single step of how to get started. So I'm thinking that I will pair program with e.g. Thomas to get set up.

File: __tests__/tests.js
Comment (Eric-Arellano) [inline __tests__/tests.js:1]: I want to add other tests, including making sure the city dropdown has all expected entries. But those can be a followup PR.

File: __tests__/tests.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+const puppeteer = require("puppeteer");
+const { beforeAll, expect, test } = require("@jest/globals");
+
+beforeAll(async () => {
```
Comment (Eric-Arellano) [inline __tests__/tests.js:4]: Here, we eagerly check that the developer ran `npm start` in another tab already. That allows us to give a good error message if they forgot to.

File: package-lock.json
Comment (Eric-Arellano) [inline package-lock.json:1]: This is a generated lockfile. Doesn't really require manual review.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "parking-lot-map",
+  "version": "1.0.0",
+  "scripts": {
+    "start": "http-server",
+    "test": "jest",
+    "fmt": "prettier --write __tests__/; eslint --fix __tests__/",
+    "lint": "prettier --check __tests__/; eslint __tests__/"
```
Comment (Eric-Arellano) [inline package.json:8]: These are the commands from the README

**Review by Eric-Arellano — COMMENTED** [review]
GitHub Actions won't start running until after we merge this PR due to GitHub's security policy.

**Review by twjordan — APPROVED** [review]
Fantastic!

**Pull request merged**